### DISCUSSION TITLE
Getting tautulli mobile apps to work, bypass oauth, traefik.

### DISCUSSION THREAD

**Discussion by Scales82** [htpcBeginner/docker-traefik#310] [Support]
Hi Guys,
Im trying to get the tautulli remote app to register with my server. However no matter what I do with the headers and custom headers in the app, I cannot get it to authenticate and bypass the oAuth settings. my current labels for Tautulli are.

```
    labels:
      - "traefik.enable=true"
      ## HTTP Routers Auth Bypass
      - "traefik.http.routers.tautulli-rtr-bypass.entrypoints=https"
      - "traefik.http.routers.tautulli-rtr-bypass.rule=Host(`tautulli.$DOMAINNAME`) && Query(`apikey`, `$TAUTULLI_API_KEY`)"
      - "traefik.http.routers.tautulli-rtr-bypass.priority=100"
      ## HTTP Routers
      - "traefik.http.routers.tautulli-rtr.entrypoints=https"
      - "traefik.http.routers.tautulli-rtr.rule=Host(`tautulli.$DOMAINNAME`)"
      ## Middlewares
      - "traefik.http.routers.tautulli-rtr.middlewares=chain-oauth@file"
      - "traefik.http.routers.tautulli-rtr-bypass.middlewares=chain-no-auth@file"
      ## HTTP Services
      - "traefik.http.routers.tautulli-rtr.service=tautulli-svc"
      - "traefik.http.routers.tautulli-rtr-bypass.service=tautulli-svc"
      - "traefik.http.services.tautulli-svc.loadbalancer.server.port=8181"
```

Please help, Id love to get the mobile app working while the main service remains secure behind oAuth

**Comment by robflate** [comment]
Mine looks like this. I just opened the Tautulli iOS app and I can successfully connect whilst still having OAuth on the main service. Some of the headers are probably redundant and I can't remember why I needed to bypass `/image` but I did it a long time ago and don't really use Tautulli anymore. I vaguely remember the old Tautulli iOS app used to connect using the API key like some 3rd party apps but it now uses a Device Key. I might be misremembering though;

```
 labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.tautulli-rtr.entrypoints=https"
      - "traefik.http.routers.tautulli-rtr.rule=Host(`tautulli.$DOMAINNAME`)"
      - "traefik.http.routers.tautulli-rtr.priority=99"
      ## Middlewares
      - "traefik.http.routers.tautulli-rtr.middlewares=chain-oauth@file"
      ## HTTP Services
      - "traefik.http.routers.tautulli-rtr.service=tautulli-svc"
      - "traefik.http.services.tautulli-svc.loadbalancer.server.port=8181"
      ## HTTP Routers Auth Bypass
      - "traefik.http.routers.tautulli-rtr-bypass.entrypoints=https"
      - "traefik.http.routers.tautulli-rtr-bypass.rule=Host(`tautulli.$DOMAINNAME`) && (Headers(`X-Api-Key`, `$TAUTULLI_DEVICE_KEY`) || Query(`apikey`, `$TAUTULLI_DEVICE_KEY`) || Headers(`X-Api-Key`, `$TAUTULLI_API_KEY`) || Query(`apikey`, `$TAUTULLI_API_KEY`) || PathPrefix(`/image`))"
      - "traefik.http.routers.tautulli-rtr-bypass.priority=100"
      - "traefik.http.routers.tautulli-rtr-bypass.service=tautulli-svc"
      - "traefik.http.routers.tautulli-rtr-bypass.middlewares=chain-no-auth@file"
```

**Reply by Scales82** [reply]
No you're right, it does have a device key thing when you go to register the app. I cant get past that point though. It allows for custom headers but I've no idea what to enter in the fields.

EDIT: Got it to work!

What I had to do was intially disable oauth and go with the no-auth middlewares. recreated the container. Registered the ios app with tautulli. once that worked and everything was operating as it should... i copied the device key and put it in the environment variables. changed the labels to what you have... particularly..
`    - "traefik.http.routers.tautulli-rtr-bypass.rule=Host(`tautulli.$DOMAINNAME`) && (Headers(`X-Api-Key`, `$TAUTULLI_DEVICE_KEY`) || Query(`apikey`, `$TAUTULLI_DEVICE_KEY`) || Headers(`X-Api-Key`, `$TAUTULLI_API_KEY`) || Query(`apikey`, `$TAUTULLI_API_KEY`) || PathPrefix(`/image`))"`

once I did that, i re-enabled oauth, recreated the container and the app still connects! 
Thanks for the assist.

**Reply by tehniemer** [reply]
I did this and am unable to connect after re-enabling authelia, not sure if that behaves differently than oauth. Any ideas? I was able to get my device registered without auth.

Here's my labels
```
   labels:
      - 'traefik.enable=true'
      ## HTTP Routers Auth Bypass
      - 'traefik.http.routers.tautulli-rtr-bypass.entrypoints=https'
      - 'traefik.http.routers.tautulli-rtr-bypass.rule=Host(tautulli.$DOMAINNAME) && (Headers(X-Api-Key, $TAUTULLI_DEVICE_KEY) || Query(apikey, $TAUTULLI_DEVICE_KEY) || Headers(X-Api-Key, $TAUTULLI_API_KEY) || Query(apikey, $TAUTULLI_API_KEY))'
      - 'traefik.http.routers.tautulli-rtr-bypass.priority=100'
      ## HTTP Routers Auth
      - 'traefik.http.routers.tautulli-rtr.entrypoints=https'
      - 'traefik.http.routers.tautulli-rtr.rule=Host(`tautulli.$DOMAINNAME`)'
      - 'traefik.http.routers.tautulli-rtr.priority=99'
      ## Middlewares
      - 'traefik.http.routers.tautulli-rtr-bypass.middlewares=chain-no-auth@file' #No Authentication
      - 'traefik.http.routers.tautulli-rtr.middlewares=chain-authelia@file' #Authelia Authentication
      ## HTTP Services
      - 'traefik.http.routers.tautulli-rtr.service=tautulli-svc'
      - 'traefik.http.services.tautulli-svc.loadbalancer.server.port=$TAUTULLI_PORT'
  ```
